### PR TITLE
[Feat] Automate smart contract docgen and PR branch creation into Linea docs repo

### PR DESCRIPTION
**Pull request by kyzooghost**
This PR implements issue(s) #4031

### Checklist

* [x] I wrote new tests for my new core changes.
* [x] I have successfully ran tests, style checker and build against my new changes locally.
* [x] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

**Review by alainncls — DISMISSED** [review]

File: contracts/docs/scripts/create-docs-website-pr-branch.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+### SCRIPT START
+
+# Ensure that `contracts` is the starting working directory
+MONOREPO_ROOT_PATH=$(git rev-parse --show-toplevel)
+cd $MONOREPO_ROOT_PATH
+cd contracts
+
+# Docgen
+npx hardhat docgen
```
Comment (thedarkjester) [inline contracts/docs/scripts/create-docs-website-pr-branch.sh:51]: do we need a step beforehand for `pnpm install` to be sure all dependencies are installed? same with Foundry - e.g. I have just pulled the repo and want the docs, how does it work cleanly?

Comment (kyzooghost) [reply contracts/docs/scripts/create-docs-website-pr-branch.sh:51]: Good point, have added a section in this script to check for installations of hardhat and foundry

File: contracts/docs/scripts/create-docs-website-pr-branch.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@
 ### ASSUMPTIONS
 # - Requires permissions to create a branch on docs.linea.build repo
 # - Must execute this script from within linea-monorepo (can be anywhere)
+# - Hardhat must be installed in the local project, forge must be installed globally
```
Comment (thedarkjester) [inline contracts/docs/scripts/create-docs-website-pr-branch.sh]:
```suggestion
# - Hardhat must be installed in the local project, Foundry must be installed globally
```
```suggestion
# - Hardhat must be installed in the local project, forge must be installed globally
```

Comment (thedarkjester) [inline contracts/docs/scripts/create-docs-website-pr-branch.sh]:
```suggestion
# - Hardhat must be installed in the local project, Foundry must be installed globally
```

**Review by thedarkjester — APPROVED** [review]

**Pull request merged**